proposals/0001-dep-process.md

```diff
@@ -0,0 +1,312 @@
+[submit]: #submit
+
+As a first step, before drafting a DEP or implementing experimental new
+protocol features, it's helpful to informally pitch your idea to see if others
+in the community are already thinking something similar, or have discussed the
+same idea in the past. This discussion could happen over chat, github issues,
+blog posts, or other channels. If you can recruit collaborators and work out
+some of the details, all the better. This period could be called **pre-DEP**.
+
+Once your idea has been flushed out, the process for proposing and debating a
+new DEP is:
+
+1. Use git to fork the [datprotocol/deps](https://github.com/datprotocol/deps)
+   repository
+1. Copy `0000-template.md` to `proposals/0000-my-proposal.md` (don't chose the
+   "next" number, use zero; `my-proposal` should be a stub identifier for the
+   proposal)
+1. Fill in the DEP template. All proposals should have a Type and Status (see
+   below for details). Feel free to tweak or expand the structure (headers,
+   content) of the document to fit your needs, but your proposal should be
+   "complete" before submission.
+1. You can submit an informal WIP (work in progress) PR whenever you like for
+   early feedback and discussion, but there is no expectation that your
+   proposal will be given detailed review until it is complete.
+1. When you are ready, submit your complete proposal for review (this could be
+   opening a PR or removing WIP status from an existing one). An editor will
+   look over your proposal for completeness; if acceptable, they will assign
+   one or more reviewers. At this stage, there are two primary outcomes: your
+   proposal is merged with "Draft", or declined. Submited proposals are
+   expected to be complete, understandable, and relevent; see below for more
+   details. This early stage of the review is expected to take **3 weeks at
+   most** from when reviewers were assigned. It is appropriate to propose
+   specific community members to review your proposal. The submiter can
+   withdraw a proposal at any time. If accepted, a DEP number will be assigned
+   and the PR merged. If there is unambiguous consensus (or, eg, a DEP is
+   documenting already adopted practice), a DEP can move directly to Active at
+   this stage.
+1. While in draft status, proposals can be experimented with. Small corrections
+   and clarifications can be submitted by PR expect to be merged quickly if
+   they are reasonable and don't change the broad behavior or semantics of the
+   proposal; larger changes should be re-submitted as Superceding proposals.
+1. When it seems approrpriate, a PR can be submitted to upgrade the status of a
+   Draft to Active. At this time a final review will take place, with the
+   outcome being that a prosal stays a Draft or is Active. It's also possible
+   for a Draft to be Closed (usuall by a specific PR to propose this). This
+   review period is shorter (**2 weeks maximum**), as the relevant reviewers
```

Comment:
Should we also make sure this includes a discussion at the next working group?

Reply:
That sounds reasonable. I included a time-bound so PRs wouldn't get "stuck", but it could be a bit longer to ensure there is time for a WG meeting. I can imagine that, at least to start, a common workflow is going to be the WG itself discussing and proposing that a DEP be stabilized to "Active" anyways, which would make the scheduling easier.

I am unsure how much WG-internals should laid out in this DEP... but I don't want to just kick those decisions down the road either.